packages/transactions/src/coders/transaction.ts

```diff
@@ -27,22 +28,27 @@ export enum TransactionType /* u8 */ {
   Upgrade = 3,
   Upload = 4,
   Blob = 5,
+  Unknown = 6,
```

Comment:
That's a good point

Reply:
I've undone this change, for two reasons: 

1) It's a reasonable assumption that there will never be a Transaction encoding that will ever begin with `-1` , certainly not intentionally.

2) that collusion is actually desirable, because the presumption is that a new transaction type would actually begin with the hex encoded `6` as that's the next available integer.

606bae5c6

Reply:
Are we making an assumption that there will be a new transaction variant of type `6` for _all_ unknown transaction types? Might it be worthwhile confirming this assumption as that's not how I read the issue.

What happens when we add a new transaction variant for `ScriptV2`? This could be type `7` - will this not throw an error?

Reply:
The assumption is the variants would be of type 6 **or greater**. 

The type when deserialized would be either `6` or greater, in which both cases when decoded [would thrown an error](https://github.com/FuelLabs/fuels-ts/pull/3003/files#diff-09abe134d06fc6102e5aa90ea9bf92c090119ee9aca9f9a9000dcae7827d94c4R699-R700) but that error is suppressed but the methods the consumer interacts with i.e.

- `getTransaction`
- `getTransactions`
- `sendTransaction`

Reply:
I was under the impression that the `UnknownTransaction` type is intended to serve as a placeholder when the TypeScript SDK cannot identify the type of a given transaction. 

Based on this, @petertonysmith94's suggestion makes sense—using the type as `-1` seems like the appropriate approach.

Reply:
The `TransactionType` [is decoded](https://github.com/FuelLabs/fuels-ts/blob/mc/chore/handle-unknown-txns/packages/transactions/src/coders/transaction.ts#L671-L672) from the serialized transaction, what reason do you have to believe that the Fuel specification would intentionally encode  `-1` as a transaction type in the future? `6` is far more likely.

Reply:
I'm under the same impression as @Torres-ssf - that unknown type was just a placeholder and not going to added to the specification with a type of 6.

My understand is that a new `ScriptV2` _could_ be added with type `6`, then another called `CreateV2` could be added with type `7` and so on... I feel we need to test these assumptions and clear up the disparity between our understandings.

Reply:
> The transaction type when deserialized would be either `6` or greater, in which both cases when decoded [would thrown an error](https://github.com/FuelLabs/fuels-ts/pull/3003/files#diff-09abe134d06fc6102e5aa90ea9bf92c090119ee9aca9f9a9000dcae7827d94c4R699-R700) but that error is suppressed by the methods the consumer interacts with i.e.
> - `getTransaction`
> - `getTransactions`
> - `sendTransaction`

Are we not just delaying the surfacing of these errors?

For example, with the `sendTransaction`, we still return a `TransactionResponse`. If the consumer is calling `getTransactionSummary` then this would lead to an error being thrown would it not?